### PR TITLE
Add breakdown for Boneshatter self damage

### PR DESCRIPTION
**Pull request by Lilylicious**
### Description of the problem being solved:
Adds a self damage breakdown to Boneshatter similar to Forbidden Rite. It takes into account all elements (phys taken as X), armour, resistances, fortify/berserk/etc.

### After screenshot:
image

### CODE REVIEW

**Comment by Lilylicious** [comment]
This PR has some overlap with #4705 in the CalcDefence file.

File: src/Modules/CalcDefence.lua [hunk 1]
```diff
@@ -60,9 +60,8 @@ function calcs.defence(env, actor)
 
 	-- Resistances
 	output.DamageReductionMax = modDB:Override(nil, "DamageReductionMax") or data.misc.DamageReductionCap
-	output.PhysicalDamageReduction = m_min(m_max(0, modDB:Sum("BASE", nil, "PhysicalDamageReduction")), output.DamageReductionMax)
-	output.PhysicalDamageReductionwhenHit = m_min(m_max(0, output.PhysicalDamageReduction + modDB:Sum("BASE", nil, "PhysicalDamageReductionWhenHit")), output.DamageReductionMax)
-	modDB:NewMod("ArmourAppliesToPhysicalDamageTaken", "FLAG", true)
```
Comment (Regisle) [inline src/Modules/CalcDefence.lua:65]: This breaks armour calcs for phys in calc defences

**Comment by Lilylicious** [comment]
Updated to work with #4673, but still has duplicated code from #4705.

**Comment by Lilylicious** [comment]
No more duplicated code, ready to merge.

**Comment by QuickStick123** [comment]
This fixes part of https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues/3063

**Review by Wires77 — APPROVED** [review]
This and Forbidden Rite seem like a great opportunity for breaking CalcDefence.lua up into a callable component so we don't have to worry about code duplication and maintenance headaches in the future.

**Pull request merged**